File: _includes/code/java-v6/src/test/java/ManageObjectsDeleteTest.java
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.WeaviateObject;
import io.weaviate.client6.v1.api.collections.query.QueryResponse;
import io.weaviate.client6.v1.api.collections.query.Where;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

import static org.assertj.core.api.Assertions.assertThat;

class ManageObjectsDeleteTest {

  private static WeaviateClient client;
  private static final String COLLECTION_NAME = "EphemeralObject";

  @BeforeAll
  public static void beforeAll() throws IOException {
    String openaiApiKey = System.getenv("OPENAI_API_KEY");
    client = WeaviateClient.connectToLocal(
        config -> config.setHeaders(Map.of("X-OpenAI-Api-Key", openaiApiKey)));
  }

  @AfterAll
  public static void afterAll() throws Exception {
    client.close();
  }

  @BeforeEach
  public void setup() throws IOException {
    if (client.collections.exists(COLLECTION_NAME)) {
      client.collections.delete(COLLECTION_NAME);
    }
    client.collections.create(COLLECTION_NAME, col -> col.properties(Property.text("name")));
  }

  @AfterEach
  public void cleanup() throws IOException {
    if (client.collections.exists(COLLECTION_NAME)) {
      client.collections.delete(COLLECTION_NAME);
    }
  }

  @Test
  void testDeleteObject() throws IOException {
    CollectionHandle<Map<String, Object>> collection = client.collections.use(COLLECTION_NAME);
    String uuidToDelete = collection.data.insert(Map.of("name", "EphemeralObjectA")).uuid();
    assertThat(collection.query.byId(uuidToDelete)).isPresent();

    // START DeleteObject
    collection.data.delete(uuidToDelete);
    // END DeleteObject

    assertThat(collection.query.byId(uuidToDelete)).isNotPresent();
  }

  @Test
  void testDeleteErrorHandling() {
    // START DeleteError
    // try {
    // String nonExistentUuid = "00000000-0000-0000-0000-000000000000";
    // CollectionHandle<Map<String, Object>> collection =
    // client.collections.use(COLLECTION_NAME);
    // collection.data.deleteById(nonExistentUuid);
    // } catch (WeaviateApiException e) {
    // // 404 error if the id was not found
    // System.out.println(e);
    // // assertThat(e.getStatusCode()).isEqualTo(404);
    // }
    // END DeleteError
  }

  @Test
  void testBatchDelete() {
    CollectionHandle<Map<String, Object>> collection = client.collections.use(COLLECTION_NAME);
    List<Map<String, Object>> objects = IntStream.range(0, 5)
        .mapToObj(i -> Map.<String, Object>of("name", "EphemeralObject_" + i))
        .collect(Collectors.toList());
    collection.data.insertMany(objects.toArray(new Map[0]));
    assertThat(collection.aggregate.overAll(a -> a.includeTotalCount(true)).totalCount()).isEqualTo(5);

    // START DeleteBatch
    collection.data.deleteMany(
        // highlight-start
        Where.property("name").like("EphemeralObject*")
    // highlight-end
    );
    // END DeleteBatch

    assertThat(collection.aggregate.overAll(a -> a.includeTotalCount(true)).totalCount()).isZero();
  }

  @Test
  void testDeleteContains() {
    // START DeleteContains
    CollectionHandle<Map<String, Object>> collection = client.collections.use(COLLECTION_NAME);
    collection.data.insertMany(
        Map.of("name", "asia"),
        Map.of("name", "europe"));

    collection.data.deleteMany(
        // highlight-start
        Where.property("name").containsAny("europe", "asia")
    // highlight-end
    );
    // END DeleteContains
  }

  @Test
  void testDryRun() {
    CollectionHandle<Map<String, Object>> collection = client.collections.use(COLLECTION_NAME);
    List<Map<String, Object>> objects = IntStream.range(0, 5)
        .mapToObj(i -> Map.<String, Object>of("name", "EphemeralObject_" + i))
        .collect(Collectors.toList());
    collection.data.insertMany(objects.toArray(new Map[0]));

    // START DryRun
    var result = collection.data.deleteMany(
        Where.property("name").like("EphemeralObject*"),
        // highlight-start
        c -> c.dryRun(true).verbose(true)
    // highlight-end
    );

    System.out.println(result);
    // END DryRun

    assertThat(result.matches()).isEqualTo(5);
    assertThat(collection.aggregate.overAll(a -> a.includeTotalCount(true)).totalCount()).isEqualTo(5);
  }

  // TODO[g-despot]: containsAny should take list not single string
  @Test
  void testBatchDeleteWithIDs() {
    CollectionHandle<Map<String, Object>> collection = client.collections.use(COLLECTION_NAME);
    List<Map<String, Object>> objects = IntStream.range(0, 5)
        .mapToObj(i -> Map.<String, Object>of("name", "EphemeralObject_" + i))
        .collect(Collectors.toList());
    collection.data.insertMany(objects.toArray(new Map[0]));
    assertThat(collection.aggregate.overAll(a -> a.includeTotalCount(true)).totalCount()).isEqualTo(5);

    // START DeleteByIDBatch
    QueryResponse<Map<String, Object>> queryResponse = collection.query.fetchObjects(q -> q.limit(3));
    List<String> ids = queryResponse.objects().stream()
        .map(WeaviateObject::uuid)
        .collect(Collectors.toList());

    collection.data.deleteMany(
        // highlight-start
        Where.uuid().containsAny(ids.toString()) // Delete the 3 objects
    // highlight-end
    );
    // END DeleteByIDBatch

    assertThat(collection.aggregate.overAll(a -> a.includeTotalCount(true)).totalCount()).isEqualTo(2);
  }
}

File: _includes/code/csharp/ManageObjectsDeleteTest.cs
using Xunit;
using Weaviate.Client;
using Weaviate.Client.Models;
using System;
using System.Threading.Tasks;
using System.Linq;
using System.Text.Json;

namespace WeaviateProject.Tests;

public class ManageObjectsDeleteTest : IAsyncLifetime
{
    private static readonly WeaviateClient client;
    private const string COLLECTION_NAME = "EphemeralObject";

    // Static constructor for one-time setup (like @BeforeAll)
    static ManageObjectsDeleteTest()
    {
        client = Connect.Local().GetAwaiter().GetResult();
    }

    // Runs before each test (like @BeforeEach)
    public async Task InitializeAsync()
    {
        if (await client.Collections.Exists(COLLECTION_NAME))
        {
            await client.Collections.Delete(COLLECTION_NAME);
        }
        await client.Collections.Create(new CollectionConfig
        {
            Name = COLLECTION_NAME,
            Properties = [Property.Text("name")]
        });
    }

    // Runs after each test (like @AfterEach and @AfterAll)
    public async Task DisposeAsync()
    {
        if (await client.Collections.Exists(COLLECTION_NAME))
        {
            await client.Collections.Delete(COLLECTION_NAME);
        }
    }

    [Fact]
    public async Task TestDeleteObject()
    {
        var collection = client.Collections.Use(COLLECTION_NAME);
        var uuidToDelete = await collection.Data.Insert(new { name = "EphemeralObjectA" });
        Assert.NotNull(await collection.Query.FetchObjectByID(uuidToDelete));

        // START DeleteObject
        await collection.Data.DeleteByID(uuidToDelete);
        // END DeleteObject

        Assert.Null(await collection.Query.FetchObjectByID(uuidToDelete));
    }

    [Fact]
    public async Task TestBatchDelete()
    {
        var collection = client.Collections.Use(COLLECTION_NAME);
        var objects = Enumerable.Range(0, 5)
            .Select(i => new { name = $"EphemeralObject_{i}" })
            .ToArray(); // Creates an array T[]

        await collection.Data.InsertMany(objects);
        var initialCount = await collection.Aggregate.OverAll(totalCount: true);
        Assert.Equal(5, initialCount.TotalCount);

        // START DeleteBatch
        await collection.Data.DeleteMany(
            // highlight-start
            Filter.Property("name").Like("EphemeralObject*")
        // highlight-end
        );
        // END DeleteBatch

        var finalCount = await collection.Aggregate.OverAll(totalCount: true);
        Assert.Equal(0, finalCount.TotalCount);
    }

    [Fact]
    public async Task TestDeleteContains()
    {
        // START DeleteContains
        var collection = client.Collections.Use(COLLECTION_NAME);
        await collection.Data.InsertMany(new[]
        {
            new { name = "asia" },
            new { name = "europe" }
        });

        await collection.Data.DeleteMany(
            // highlight-start
            Filter.Property("name").ContainsAny(["europe", "asia"])
        // highlight-end
        );
        // END DeleteContains
    }

    [Fact]
    public async Task TestDryRun()
    {
        var collection = client.Collections.Use(COLLECTION_NAME);
        var objects = Enumerable.Range(0, 5)
            .Select(i => new { name = $"EphemeralObject_{i}" })
            .ToArray(); // Creates an array T[]

        await collection.Data.InsertMany(objects);

        // START DryRun
        var result = await collection.Data.DeleteMany(
            Filter.Property("name").Like("EphemeralObject*"),
            // highlight-start
            dryRun: true
        // highlight-end
        );

        Console.WriteLine(JsonSerializer.Serialize(result));
        // END DryRun

        Assert.Equal(5, result.Matches);
        var finalCount = await collection.Aggregate.OverAll(totalCount: true);
        Assert.Equal(5, finalCount.TotalCount); // Objects should not be deleted
    }

    [Fact]
    public async Task TestBatchDeleteWithIDs()
    {
        var collection = client.Collections.Use(COLLECTION_NAME);
        var objects = Enumerable.Range(0, 5)
            .Select(i => new { name = $"EphemeralObject_{i}" })
            .ToArray(); // Creates an array T[]

        await collection.Data.InsertMany(objects);
        var initialCount = await collection.Aggregate.OverAll(totalCount: true);
        Assert.Equal(5, initialCount.TotalCount);

        // START DeleteByIDBatch
        var queryResponse = await collection.Query.FetchObjects(limit: 3);
        var ids = queryResponse.Objects.Select(obj => obj.ID.Value).ToList();

        await collection.Data.DeleteMany(
            // highlight-start
            Filter.ID.ContainsAny(ids) // Delete the 3 objects
                                       // highlight-end
        );
        // END DeleteByIDBatch

        var finalCount = await collection.Aggregate.OverAll(totalCount: true);
        Assert.Equal(2, finalCount.TotalCount);
    }
}